mealie/repos/repository_recipes.py

```diff
@@ -143,3 +143,10 @@ def get_random(self, limit=1) -> list[Recipe]:
             .order_by(func.random())  # Postgres and SQLite specific
             .limit(limit)
         ]
+
+
+    def get_by_slug(self, slug: str, limit=1) -> Optional[Recipe]:
+        dbrecipe = self.session.query(RecipeModel).filter(RecipeModel.slug == slug).one_or_none()
+        if dbrecipe is None:
+            return None
+        return self.schema.from_orm(dbrecipe)
```

Comment:
You'll need to consider the group_id here since the recipe table is unique based off of group_id and the slug. This would return a results for any recipe even if it's in another group.

I don't think function is necessary because the recipes repo uses the slug as a primary key. Just be sure to call `by_group` to filter by the group.

Reply:
Ok, do you mean to pass it as parameter or use `self.group_id`?